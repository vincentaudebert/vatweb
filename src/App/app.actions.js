import constants from './app.constants';

export const changeFile = slug => {
  return {
    type: constants.CHANGE_FILE,
    payload: {
      slug,
    },
  };
};

export const closeFile = slug => {
  return {
    type: constants.CLOSE_FILE,
    payload: {
      slug,
    },
  };
};
