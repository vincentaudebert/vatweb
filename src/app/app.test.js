import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import App from './app';

jest.mock('../middle-bar/middle-bar.js', () => 'div');
jest.mock('../top-bar/top-bar.container.js', () => 'div');
jest.mock('../footer-bar/footer-bar.js', () => 'div');
jest.mock('react-ga', () => ({
  pageview: () => {},
}));

describe('App', () => {
  let changeFile;
  let match;
  let history;

  beforeEach(() => {
    changeFile = jest.fn();
    match = {
      params: {
        slug: 'me.js',
      },
    };

    history = {
      push: jest.fn(),
    };

    const div = document.createElement('div');
    window.domNode = div;
    document.body.appendChild(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<App changeFile={changeFile} match={match} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with no slug', () => {
    match = {
      params: {},
    };
    const tree = renderer
      .create(<App changeFile={changeFile} match={match} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('behaves correctly after wantedLocation update', () => {
    const app = mount(
      <App changeFile={changeFile} match={match} history={history} />
    );
    expect(changeFile).toHaveBeenCalledTimes(1);

    app.setProps({
      wantedLocation: '1-apocope.js',
    });

    expect(history.push).toHaveBeenCalledTimes(1);
  });

  it('behaves correctly after no slug update', () => {
    const app = mount(
      <App changeFile={changeFile} match={match} history={history} />
    );
    expect(changeFile).toHaveBeenCalledTimes(1);

    app.setProps({
      match: { params: {} },
    });

    expect(changeFile).toHaveBeenCalledTimes(2);
  });

  it('hide/show editor', () => {
    const app = mount(<App changeFile={changeFile} match={match} />);
    expect(app.instance().state.editorOpen).toBeTruthy();
    app.instance().toggleEditor(false);
    expect(app.instance().state.editorOpen).toBeFalsy();

    app.find('button.btn--fake').simulate('click');
    expect(app.instance().state.editorOpen).toBeTruthy();
  });

  it('hidden editor', () => {
    const app = mount(<App changeFile={changeFile} match={match} />);
    app.setState({
      editorOpen: false,
    });
    expect(app).toMatchSnapshot();
  });

  it('test animation end function', () => {
    const app = mount(<App changeFile={changeFile} match={match} />, {
      attachTo: window.domNode,
    });
    app.setState({
      editorOpen: false,
    });
    app.instance().editorAnimationEnd();
    expect(app.render()).toMatchSnapshot();
  });

  it('test animation end function not triggered if editor is open', () => {
    const app = mount(<App changeFile={changeFile} match={match} />, {
      attachTo: window.domNode,
    });
    app.setState({
      editorOpen: true,
    });
    app.instance().editorAnimationEnd();
    expect(app.render()).toMatchSnapshot();
  });
});
