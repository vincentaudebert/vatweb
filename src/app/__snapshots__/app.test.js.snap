// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App hidden editor 1`] = `
<App
  changeFile={
    [MockFunction] {
      "calls": Array [
        Array [
          "me.js",
        ],
        Array [
          "me.js",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  match={
    Object {
      "params": Object {
        "slug": "me.js",
      },
    }
  }
>
  <div
    className="App"
  >
    <div
      className="editor fade hide"
      id="editor"
    >
      <div
        toggleEditor={[Function]}
      />
      <div />
      <div />
    </div>
    <div
      className="alternative d-none fade show"
      id="alternative"
    >
      <div
        className="container"
      >
        <h2>
          Oopsie...
        </h2>
        <p>
          Don't worry. I'm not offended. I knew you would try all the buttons. And particularly the closing one :)
        </p>
        <p>
          If you want, you can reopen my life editor by clicking here:
           
          <button
            className="d-inline-block btn--fake reset-button"
            onClick={[Function]}
            title="Reopen editor"
          >
            <FaMagic>
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 512 512",
                  }
                }
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 96l16-32 32-16-32-16-16-32-16 32-32 16 32 16 16 32zM80 160l26.66-53.33L160 80l-53.34-26.67L80 0 53.34 53.33 0 80l53.34 26.67L80 160zm352 128l-26.66 53.33L352 368l53.34 26.67L432 448l26.66-53.33L512 368l-53.34-26.67L432 288zm70.62-193.77L417.77 9.38C411.53 3.12 403.34 0 395.15 0c-8.19 0-16.38 3.12-22.63 9.38L9.38 372.52c-12.5 12.5-12.5 32.76 0 45.25l84.85 84.85c6.25 6.25 14.44 9.37 22.62 9.37 8.19 0 16.38-3.12 22.63-9.37l363.14-363.15c12.5-12.48 12.5-32.75 0-45.24zM359.45 203.46l-50.91-50.91 86.6-86.6 50.91 50.91-86.6 86.6z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaMagic>
             
            <FaBackward>
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 512 512",
                  }
                }
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.5 280.6l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2zm256 0l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaBackward>
             
            <FaHandPeace>
              <IconBase
                attr={
                  Object {
                    "viewBox": "0 0 448 512",
                  }
                }
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M408 216c-22.092 0-40 17.909-40 40h-8v-32c0-22.091-17.908-40-40-40s-40 17.909-40 40v32h-8V48c0-26.51-21.49-48-48-48s-48 21.49-48 48v208h-13.572L92.688 78.449C82.994 53.774 55.134 41.63 30.461 51.324 5.787 61.017-6.356 88.877 3.337 113.551l74.765 190.342-31.09 24.872c-15.381 12.306-19.515 33.978-9.741 51.081l64 112A39.998 39.998 0 0 0 136 512h240c18.562 0 34.686-12.77 38.937-30.838l32-136A39.97 39.97 0 0 0 448 336v-80c0-22.091-17.908-40-40-40z"
                    key="0"
                  />
                </svg>
              </IconBase>
            </FaHandPeace>
          </button>
        </p>
        <p>
          If you got sick of reading some (beautiful) code, you can download my
           
          <a
            className="alternative--link"
            href="https://github.com/vincentaudebert/vatweb/raw/master/public/assets/Vincent_Audebert_CV.pdf"
            rel="noreferrer noopener"
            target="_blank"
            title="PDF CV"
          >
            CV in a PDF format
          </a>
          .
        </p>
        <p>
          Vincent.
        </p>
      </div>
    </div>
  </div>
</App>
`;

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="editor fade show"
    id="editor"
  >
    <div
      toggleEditor={[Function]}
    />
    <div />
    <div />
  </div>
  <div
    className="alternative d-none fade hide"
    id="alternative"
  >
    <div
      className="container"
    >
      <h2>
        Oopsie...
      </h2>
      <p>
        Don't worry. I'm not offended. I knew you would try all the buttons. And particularly the closing one :)
      </p>
      <p>
        If you want, you can reopen my life editor by clicking here:
         
        <button
          className="d-inline-block btn--fake reset-button"
          onClick={[Function]}
          title="Reopen editor"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 96l16-32 32-16-32-16-16-32-16 32-32 16 32 16 16 32zM80 160l26.66-53.33L160 80l-53.34-26.67L80 0 53.34 53.33 0 80l53.34 26.67L80 160zm352 128l-26.66 53.33L352 368l53.34 26.67L432 448l26.66-53.33L512 368l-53.34-26.67L432 288zm70.62-193.77L417.77 9.38C411.53 3.12 403.34 0 395.15 0c-8.19 0-16.38 3.12-22.63 9.38L9.38 372.52c-12.5 12.5-12.5 32.76 0 45.25l84.85 84.85c6.25 6.25 14.44 9.37 22.62 9.37 8.19 0 16.38-3.12 22.63-9.37l363.14-363.15c12.5-12.48 12.5-32.75 0-45.24zM359.45 203.46l-50.91-50.91 86.6-86.6 50.91 50.91-86.6 86.6z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 280.6l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2zm256 0l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M408 216c-22.092 0-40 17.909-40 40h-8v-32c0-22.091-17.908-40-40-40s-40 17.909-40 40v32h-8V48c0-26.51-21.49-48-48-48s-48 21.49-48 48v208h-13.572L92.688 78.449C82.994 53.774 55.134 41.63 30.461 51.324 5.787 61.017-6.356 88.877 3.337 113.551l74.765 190.342-31.09 24.872c-15.381 12.306-19.515 33.978-9.741 51.081l64 112A39.998 39.998 0 0 0 136 512h240c18.562 0 34.686-12.77 38.937-30.838l32-136A39.97 39.97 0 0 0 448 336v-80c0-22.091-17.908-40-40-40z"
            />
          </svg>
        </button>
      </p>
      <p>
        If you got sick of reading some (beautiful) code, you can download my
         
        <a
          className="alternative--link"
          href="https://github.com/vincentaudebert/vatweb/raw/master/public/assets/Vincent_Audebert_CV.pdf"
          rel="noreferrer noopener"
          target="_blank"
          title="PDF CV"
        >
          CV in a PDF format
        </a>
        .
      </p>
      <p>
        Vincent.
      </p>
    </div>
  </div>
</div>
`;

exports[`App renders correctly with no slug 1`] = `
<div
  className="App"
>
  <div
    className="editor fade show"
    id="editor"
  >
    <div
      toggleEditor={[Function]}
    />
    <div />
    <div />
  </div>
  <div
    className="alternative d-none fade hide"
    id="alternative"
  >
    <div
      className="container"
    >
      <h2>
        Oopsie...
      </h2>
      <p>
        Don't worry. I'm not offended. I knew you would try all the buttons. And particularly the closing one :)
      </p>
      <p>
        If you want, you can reopen my life editor by clicking here:
         
        <button
          className="d-inline-block btn--fake reset-button"
          onClick={[Function]}
          title="Reopen editor"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 96l16-32 32-16-32-16-16-32-16 32-32 16 32 16 16 32zM80 160l26.66-53.33L160 80l-53.34-26.67L80 0 53.34 53.33 0 80l53.34 26.67L80 160zm352 128l-26.66 53.33L352 368l53.34 26.67L432 448l26.66-53.33L512 368l-53.34-26.67L432 288zm70.62-193.77L417.77 9.38C411.53 3.12 403.34 0 395.15 0c-8.19 0-16.38 3.12-22.63 9.38L9.38 372.52c-12.5 12.5-12.5 32.76 0 45.25l84.85 84.85c6.25 6.25 14.44 9.37 22.62 9.37 8.19 0 16.38-3.12 22.63-9.37l363.14-363.15c12.5-12.48 12.5-32.75 0-45.24zM359.45 203.46l-50.91-50.91 86.6-86.6 50.91 50.91-86.6 86.6z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 280.6l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2zm256 0l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M408 216c-22.092 0-40 17.909-40 40h-8v-32c0-22.091-17.908-40-40-40s-40 17.909-40 40v32h-8V48c0-26.51-21.49-48-48-48s-48 21.49-48 48v208h-13.572L92.688 78.449C82.994 53.774 55.134 41.63 30.461 51.324 5.787 61.017-6.356 88.877 3.337 113.551l74.765 190.342-31.09 24.872c-15.381 12.306-19.515 33.978-9.741 51.081l64 112A39.998 39.998 0 0 0 136 512h240c18.562 0 34.686-12.77 38.937-30.838l32-136A39.97 39.97 0 0 0 448 336v-80c0-22.091-17.908-40-40-40z"
            />
          </svg>
        </button>
      </p>
      <p>
        If you got sick of reading some (beautiful) code, you can download my
         
        <a
          className="alternative--link"
          href="https://github.com/vincentaudebert/vatweb/raw/master/public/assets/Vincent_Audebert_CV.pdf"
          rel="noreferrer noopener"
          target="_blank"
          title="PDF CV"
        >
          CV in a PDF format
        </a>
        .
      </p>
      <p>
        Vincent.
      </p>
    </div>
  </div>
</div>
`;

exports[`App test animation end function 1`] = `
<div
  class="App"
>
  <div
    class="editor fade hide d-none"
    id="editor"
  >
    <div />
    <div />
    <div />
  </div>
  <div
    class="alternative fade show d-block"
    id="alternative"
  >
    <div
      class="container"
    >
      <h2>
        Oopsie...
      </h2>
      <p>
        Don't worry. I'm not offended. I knew you would try all the buttons. And particularly the closing one :)
      </p>
      <p>
        If you want, you can reopen my life editor by clicking here: 
        <button
          class="d-inline-block btn--fake reset-button"
          title="Reopen editor"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 96l16-32 32-16-32-16-16-32-16 32-32 16 32 16 16 32zM80 160l26.66-53.33L160 80l-53.34-26.67L80 0 53.34 53.33 0 80l53.34 26.67L80 160zm352 128l-26.66 53.33L352 368l53.34 26.67L432 448l26.66-53.33L512 368l-53.34-26.67L432 288zm70.62-193.77L417.77 9.38C411.53 3.12 403.34 0 395.15 0c-8.19 0-16.38 3.12-22.63 9.38L9.38 372.52c-12.5 12.5-12.5 32.76 0 45.25l84.85 84.85c6.25 6.25 14.44 9.37 22.62 9.37 8.19 0 16.38-3.12 22.63-9.37l363.14-363.15c12.5-12.48 12.5-32.75 0-45.24zM359.45 203.46l-50.91-50.91 86.6-86.6 50.91 50.91-86.6 86.6z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 280.6l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2zm256 0l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M408 216c-22.092 0-40 17.909-40 40h-8v-32c0-22.091-17.908-40-40-40s-40 17.909-40 40v32h-8V48c0-26.51-21.49-48-48-48s-48 21.49-48 48v208h-13.572L92.688 78.449C82.994 53.774 55.134 41.63 30.461 51.324 5.787 61.017-6.356 88.877 3.337 113.551l74.765 190.342-31.09 24.872c-15.381 12.306-19.515 33.978-9.741 51.081l64 112A39.998 39.998 0 0 0 136 512h240c18.562 0 34.686-12.77 38.937-30.838l32-136A39.97 39.97 0 0 0 448 336v-80c0-22.091-17.908-40-40-40z"
            />
          </svg>
        </button>
      </p>
      <p>
        If you got sick of reading some (beautiful) code, you can download my 
        <a
          class="alternative--link"
          href="https://github.com/vincentaudebert/vatweb/raw/master/public/assets/Vincent_Audebert_CV.pdf"
          rel="noreferrer noopener"
          target="_blank"
          title="PDF CV"
        >
          CV in a PDF format
        </a>
        .
      </p>
      <p>
        Vincent.
      </p>
    </div>
  </div>
</div>
`;

exports[`App test animation end function not triggered if editor is open 1`] = `
<div
  class="App"
>
  <div
    class="editor fade show"
    id="editor"
  >
    <div />
    <div />
    <div />
  </div>
  <div
    class="alternative d-none fade hide"
    id="alternative"
  >
    <div
      class="container"
    >
      <h2>
        Oopsie...
      </h2>
      <p>
        Don't worry. I'm not offended. I knew you would try all the buttons. And particularly the closing one :)
      </p>
      <p>
        If you want, you can reopen my life editor by clicking here: 
        <button
          class="d-inline-block btn--fake reset-button"
          title="Reopen editor"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 96l16-32 32-16-32-16-16-32-16 32-32 16 32 16 16 32zM80 160l26.66-53.33L160 80l-53.34-26.67L80 0 53.34 53.33 0 80l53.34 26.67L80 160zm352 128l-26.66 53.33L352 368l53.34 26.67L432 448l26.66-53.33L512 368l-53.34-26.67L432 288zm70.62-193.77L417.77 9.38C411.53 3.12 403.34 0 395.15 0c-8.19 0-16.38 3.12-22.63 9.38L9.38 372.52c-12.5 12.5-12.5 32.76 0 45.25l84.85 84.85c6.25 6.25 14.44 9.37 22.62 9.37 8.19 0 16.38-3.12 22.63-9.37l363.14-363.15c12.5-12.48 12.5-32.75 0-45.24zM359.45 203.46l-50.91-50.91 86.6-86.6 50.91 50.91-86.6 86.6z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 280.6l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2zm256 0l192 160c20.6 17.2 52.5 2.8 52.5-24.6V96c0-27.4-31.9-41.8-52.5-24.6l-192 160c-15.3 12.8-15.3 36.4 0 49.2z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M408 216c-22.092 0-40 17.909-40 40h-8v-32c0-22.091-17.908-40-40-40s-40 17.909-40 40v32h-8V48c0-26.51-21.49-48-48-48s-48 21.49-48 48v208h-13.572L92.688 78.449C82.994 53.774 55.134 41.63 30.461 51.324 5.787 61.017-6.356 88.877 3.337 113.551l74.765 190.342-31.09 24.872c-15.381 12.306-19.515 33.978-9.741 51.081l64 112A39.998 39.998 0 0 0 136 512h240c18.562 0 34.686-12.77 38.937-30.838l32-136A39.97 39.97 0 0 0 448 336v-80c0-22.091-17.908-40-40-40z"
            />
          </svg>
        </button>
      </p>
      <p>
        If you got sick of reading some (beautiful) code, you can download my 
        <a
          class="alternative--link"
          href="https://github.com/vincentaudebert/vatweb/raw/master/public/assets/Vincent_Audebert_CV.pdf"
          rel="noreferrer noopener"
          target="_blank"
          title="PDF CV"
        >
          CV in a PDF format
        </a>
        .
      </p>
      <p>
        Vincent.
      </p>
    </div>
  </div>
</div>
`;
